packages/cli-hermes/src/profileHermes/sourcemapUtils.ts

```diff
@@ -0,0 +1,120 @@
+import {Config} from '@react-native-community/cli-types';
+import {logger, CLIError} from '@react-native-community/cli-tools';
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+import {SourceMap} from 'hermes-profile-transformer';
+import ip from 'ip';
+import nodeFetch from 'node-fetch';
+
+function getTempFilePath(filename: string) {
+  return path.join(os.tmpdir(), filename);
+}
+
+function writeJsonSync(targetPath: string, data: any) {
+  let json;
+  try {
+    json = JSON.stringify(data);
+  } catch (e) {
+    logger.error(
+      `Failed to serialize data to json before writing to ${targetPath}\n`,
+      e,
+    );
```

Comment:
I think we should throw CLIError here and in line 28. Currently an undefined `json` will be passed to `fs.writeFileSync` in case of parsing failure and that doesn't make too much sense, isn't it?

Reply:
Yup. Let's stick to the same error handling everywhere - rethrowing CLIErrors, while not perfect, provides a unified error handling experience. 

PS. It is okay to provide additional logging (like logger.info/error) when needed, for better UX.